tools/pydantic_codegen/plugin.py

```diff
@@ -0,0 +1,387 @@
+IMPORT_ANNOTATIONS = generate_import_from(["annotations"], from_="__future__")
+# `from .base import GQLBase, Typename, etc.`
+IMPORT_CUSTOM_TYPES = generate_import_from([GQL_BASE, TYPENAME], from_="base", level=1)
+# `import sys`
+IMPORT_SYS = generate_import(["sys"])
+
+
+#: Names that must be conditionally imported from `typing` or `typing_extensions` depending on python version.
+TYPING_IMPORTS_TO_REWRITE: Final[tuple[str, ...]] = (
+    "override",
+    "Annotated",
+    "Literal",
+)
+
+
+class FixFragmentOrder(Plugin):
+    """Codegen plugin to fix inconsistent ordering of fragments module.
+
+    HACK: At the time of implementation, the fragments module has inconsistent ordering of
+    - class definitions
+    - `Class.model_rebuild()` statements
+
+    See: https://github.com/mirumee/ariadne-codegen/issues/315. This plugin is a workaround in the meantime.
+    """
+
+    def generate_fragments_module(self, module: ast.Module, *_, **__) -> ast.Module:
+        return self._ensure_class_order(module)
+
+    @staticmethod
+    def _ensure_class_order(module: ast.Module) -> ast.Module:
+        # Separate the statements into the following expected groups:
+        # - imports
+        # - class definitions
+        # - Model.model_rebuild() statements
+        imports: deque[ast.Import | ast.ImportFrom] = deque()
+        class_defs: deque[ast.ClassDef] = deque()
+        model_rebuilds: deque[ast.Expr] = deque()
+
+        for stmt in module.body:
+            if is_import_stmt(stmt):
+                imports.append(stmt)
+            elif isinstance(stmt, ast.ClassDef):
+                class_defs.append(stmt)
+            elif is_model_rebuild_expr(stmt):
+                model_rebuilds.append(stmt)
+            else:
+                stmt_type = type(stmt).__name__
+                stmt_repr = ast.unparse(stmt)
+                raise TypeError(f"Unexpected {stmt_type!r} statement:\n{stmt_repr}")
+
+        # Deterministically reorder the class definitions, ensuring parent classes are defined first
+        classnames = sort_classnames(class_defs)
+        classname2index = {name: idx for idx, name in enumerate(classnames)}
+
+        # Reorder the class definitions and `Model.model_rebuild()` statements
+        class_defs = sorted(class_defs, key=lambda stmt: classname2index[stmt.name])
+        model_rebuilds = sorted(
+            model_rebuilds, key=lambda stmt: classname2index[stmt.value.func.value.id]
+        )
+
+        module.body = [*imports, *class_defs, *model_rebuilds]
+        return module
+
+
+def _forget_default_graphql_id_type() -> None:
+    # HACK: Override the default python type that ariadne-codegen uses for GraphQL's `ID` type.
+    # See: https://github.com/mirumee/ariadne-codegen/issues/316
+    from ariadne_codegen.client_generators import constants
+
+    with suppress(LookupError):
+        constants.SIMPLE_TYPE_MAP.pop("ID")
+        constants.INPUT_SCALARS_MAP.pop("ID")
```

Comment:
**note:** Also a deliberate hack/workaround (see code comment).  I wanted to override the behavior of ID fields so that they were
- not directly editable 
- hidden from being displayed in the printed representation (`str`/`repr`) of any of these objects